### PR TITLE
docs(self-hosted): describe production setup

### LINKED ISSUE #3985: Document Metrics Endpoint
The metrics endpoint should be integrated in the documentation.

**Acceptance criteria**
- [ ] Add metrics endpoint
- [ ] How can customers access data from the metrics
- [ ] What kind of metrics are included
- [ ] Document how users know if ZITADEL is running, healthz endpoint, readiness, etc.

### PR DESCRIPTION
**Pull request by eliobischof**
Guides going from PoC to production.

closes #3985 

Triggered by https://discord.com/channels/927474939156643850/927866013545025566/1036592427055714304

### CODE REVIEW

File: docs/docs/guides/deploy/overview.mdx [hunk 1]
```diff
@@ -3,26 +3,31 @@ title: Overview
...
 - [MacOS](./macos)
 - [Docker Compose](./compose)
 - [Knative](./knative)
 - [Kubernetes](./kubernetes)
 
-## Prerequisits
+## Prerequisites
 
-- ZITADEL does not need many resources, 1 CPU and 512MB memory are more than enough. (With more CPU, the password hashing might be faster)
+- When run as a proof of concept, ZITADEL does not need many resources, 1 CPU and 512MB memory are more than enough. (With more CPU, the password hashing might be faster)
```
Comment (fforootd) [inline docs/docs/guides/deploy/overview.mdx:15]:
```suggestion
- For test environments, ZITADEL does not need many resources, 1 CPU and 512MB memory are more than enough. (With more CPU, the password hashing might be faster)
```

File: docs/docs/guides/deploy/overview.mdx [hunk 2]
```diff
@@ -3,26 +3,31 @@ title: Overview
...
 
 ## Releases
 
 The easiest way to use ZITADEL is to run one of our container releases
 
 - ZITADEL does provide latest and stable [container images](https://github.com/zitadel/zitadel/pkgs/container/zitadel)
 - **stable** is the current **production** release of ZITADEL.
-- **latest** is the **last created** release from our pipelines that gets updated on a high frequency.
\ No newline at end of file
+- **latest** is the **last created** release from our pipelines that gets updated on a high frequency.
```
Comment (fforootd) [inline docs/docs/guides/deploy/overview.mdx:24]:
```suggestion
- **latest** is the **last created** release from our pipelines that gets updated in a high frequency.
```

File: docs/docs/guides/deploy/overview.mdx [hunk 3]
```diff
@@ -3,26 +3,31 @@ title: Overview
...
+
+# Production Setup
+
+As soon as you successfully created your first proof of concept using one of the deployment guides in this section,
```
Comment (fforootd) [inline docs/docs/guides/deploy/overview.mdx:28]:
```suggestion
As soon as you successfully created your first test environment using one of the deployment guides in this section,
```

File: docs/docs/guides/deploy/overview.mdx [hunk 4]
```diff
@@ -3,26 +3,31 @@ title: Overview
...
+you might want to configure ZITADEL for production and embed it into your system landscape.
+To do so, jump straight to the [production setup guide](../manage/self-hosted/production).
+
+For achieving high availablility, we recommend to use [Kubernetes](https://kubernetes.io/docs/home/).
+We have an official [Helm chart](https://artifacthub.io/packages/helm/zitadel/zitadel) for easy deployment and maintenance.
```
Comment (fforootd) [inline docs/docs/guides/deploy/overview.mdx:33]:
```suggestion
To achieving high availability, we recommend to use a [Kubernetes](https://kubernetes.io/docs/home/) Cluster.
We have an official [Helm chart](https://artifacthub.io/packages/helm/zitadel/zitadel) for easy deployment and maintenance.
```

File: docs/docs/guides/manage/self-hosted/configure/docker-compose.yaml [hunk 5]
```diff
@@ -1,53 +1,55 @@
-version: '3.8'
```
Comment (fforootd) [inline docs/docs/guides/manage/self-hosted/configure/docker-compose.yaml:1]: Are we sure we want to change single to double quotes?

File: docs/docs/guides/manage/self-hosted/configure/docker-compose.yaml [hunk 6]
```diff
@@ -1,53 +1,55 @@
...
     ports:
-      - '9090:8080'
-      - '26257:26257'
+      - "9090:8080"
+      - "26257:26257"
     volumes:
-      - 'certs:/cockroach/certs:rw'
-      - 'data:/cockroach/cockroach-data:rw'
+      - "certs:/cockroach/certs:rw"
+      - "data:/cockroach/cockroach-data:rw"
```
Comment (fforootd) [inline docs/docs/guides/manage/self-hosted/configure/docker-compose.yaml:52]: Lets also add the CRDB data dir for persistency here

File: docs/docs/guides/manage/self-hosted/production.md [hunk 7]
```diff
@@ -0,0 +1,102 @@
+---
+title: Production Checklist
+---
+
+As soon as you successfully deployed ZITADEL as a proof of concept using one of our [deployment guides](/docs/guides/deploy/overview),
+you are ready to configure ZITADEL for production use in your system landscape.
```
Comment (fforootd) [inline docs/docs/guides/manage/self-hosted/production.md:6]:
```suggestion
you are ready to configure ZITADEL for production usage.
```

File: docs/docs/guides/manage/self-hosted/production.md [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+
+## High Availability
+
+We recommend running ZITADEL higly available using an orchestrator that schedules ZITADEL on multiple nodes, like [Kubernetes](/docs/guides/deploy/kubernetes).
```
Comment (fforootd) [inline docs/docs/guides/manage/self-hosted/production.md:10]:
```suggestion
We recommend running ZITADEL highly available using an orchestrator that schedules ZITADEL on multiple server, like [Kubernetes](/docs/guides/deploy/kubernetes).
```

File: docs/docs/guides/manage/self-hosted/production.md [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+
+## Binary Configuration
```
Comment (fforootd) [inline docs/docs/guides/manage/self-hosted/production.md:12]:
```suggestion
## Configuration
```

File: docs/docs/guides/manage/self-hosted/production.md [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+
+Read [on the configure page](/docs/guides/manage/self-hosted/configure) about the available options you have to configure the ZITADEL binary.
```
Comment (fforootd) [inline docs/docs/guides/manage/self-hosted/production.md:14]:
```suggestion
Read [on the configure page](/docs/guides/manage/self-hosted/configure) about the available options you have to configure the ZITADEL.
```

File: docs/docs/guides/manage/self-hosted/production.md [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+
+## Networking
+
+- For enabling access via **HTTPS** and restricting access to **HTTPS**, head to [the description of your TLS options](/docs/guides/manage/self-hosted/tls_modes).
+- If you want to front ZITADEL with a reverse proxy, web application firewall or content delivery network, make sure to support **[HTTP/2](/docs/guides/manage/self-hosted/http2)**.
+- You can also refer to some **[example reverse proxy configurations](/docs/guides/manage/self-hosted/reverseproxy/reverse_proxy)**.
```
Comment (fforootd) [inline docs/docs/guides/manage/self-hosted/production.md:20]:
```suggestion
- To enable and restrict access to **HTTPS**, head over to [the description of your TLS options](/docs/guides/manage/self-hosted/tls_modes).
- If you want to front ZITADEL with a reverse proxy, web application firewall or content delivery network, make sure to support **[HTTP/2](/docs/guides/manage/self-hosted/http2)**.
- You can also refer to some **[example reverse proxy configurations](/docs/guides/manage/self-hosted/reverseproxy/reverse_proxy)**.
```

File: docs/docs/guides/manage/self-hosted/production.md [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+
+## Monitoring
+
+By default, **metrics** are exposed at /debug/metrics in OpenTelemetry (otel) format.
+They can easily be consumed by an [otel collector](https://opentelemetry.io/docs/collector/) and be sent to or scraped by various backends,
+for example [Prometheus](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/prometheusexporter).
```
Comment (fforootd) [inline docs/docs/guides/manage/self-hosted/production.md:26]: Please also point to https://docs.zitadel.com/docs/apis/observability/metrics

File: docs/docs/guides/manage/self-hosted/production.md [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+Tracing:
+  # Choose one in "otel", "google", "log" and "none"
+  Type: google
+  Fraction: 1
+  MetricPrefix: zitadel
+```
+
+## Database
+
+Depending on your environment, you maybe would want to tweak some settings about how ZITADEL interacts with the database in the Database section of your ZITADEL configuration. Read more about your [database configuration options](/docs/guides/manage/self-hosted/database).
```
Comment (fforootd) [inline docs/docs/guides/manage/self-hosted/production.md:40]:
```suggestion
Depending on your environment, you maybe would want to tweak some settings about how ZITADEL interacts with the database in the database section of your ZITADEL configuration. Read more about your [database configuration options](/docs/guides/manage/self-hosted/database).
```

File: docs/docs/guides/manage/self-hosted/production.md [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+    Database: zitadel
+    //highlight-start
+    MaxOpenConns: 20
+    MaxConnLifetime: 30m
+    MaxConnIdleTime: 30m
+    //highlight-end
+    Options: ""
+```
+
+You also might want to configure how projections are computed. These are the default values:
```
Comment (fforootd) [inline docs/docs/guides/manage/self-hosted/production.md:56]: I think we need to point people to a concepts page to explain what projections are

File: docs/docs/guides/manage/self-hosted/configure/docker-compose.yaml (hunk 5 above)
Comment (eliobischof) [reply docs/docs/guides/manage/self-hosted/configure/docker-compose.yaml:1]: My vscode linter did that 🤷‍♂️😬

File: docs/docs/guides/deploy/overview.mdx (hunk 4 above)
Comment (eliobischof) [reply docs/docs/guides/deploy/overview.mdx:33]: I like availabilility 😄

File: docs/docs/guides/manage/self-hosted/production.md (hunk 14 above)
Comment (eliobischof) [reply docs/docs/guides/manage/self-hosted/production.md:56]: Will do 👍

File: docs/docs/guides/manage/self-hosted/production.md (hunk 12 above)
Comment (eliobischof) [reply docs/docs/guides/manage/self-hosted/production.md:26]: Will do 👍

File: docs/docs/guides/manage/self-hosted/configure/docker-compose.yaml (hunk 6 above)
Comment (eliobischof) [reply docs/docs/guides/manage/self-hosted/configure/docker-compose.yaml:52]: /cockroach/cockroach-data is the data dir... Am I missing something?

File: docs/docs/guides/manage/self-hosted/configure/docker-compose.yaml (hunk 5 above)
Comment (eliobischof) [reply docs/docs/guides/manage/self-hosted/configure/docker-compose.yaml:1]: I think it doesn't make a big difference, but [escaping characters is more similar with other languages when using double quotes](http://yaml.org/spec/1.2-old/spec.html#id2776092). Therefore, double quotes are slightly better, IMO.

Comment (fforootd) [reply docs/docs/guides/manage/self-hosted/configure/docker-compose.yaml:1]: in that case i think we should set up a linter for yaml stuff.

I am fine with using doubles if we stick to it 😁

**Review by fforootd — APPROVED** [review]

File: docs/docs/guides/manage/self-hosted/configure/docker-compose.yaml (hunk 5 above)
Comment (eliobischof) [reply docs/docs/guides/manage/self-hosted/configure/docker-compose.yaml:1]: Added formatting for all **.md files and everything in the /docs folder.

**Review by fforootd — COMMENTED** [review]
Why don't we split the md lint as separate pr?

I think there are things needed to discuss first (like do we want to lint the GH templates, ...)

**Comment by eliobischof** [comment]
> Why don't we split the md lint as separate pr?
> 
> I think there are things needed to discuss first (like do we want to lint the GH templates, ...)

I "moved" the relevant commit to a separate PR #4650

**Pull request merged**